pkgs/by-name/sp/spotify-backup/package.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, fetchFromGitHub
+, python3
+}:
+
+python3.pkgs.buildPythonPackage {
+  pname = "spotify-backup";
+  version = "0-unstable-2023-02-24";
+
+  src = fetchFromGitHub {
+    owner = "caseychu";
+    repo = "spotify-backup";
+    rev = "0ccf12c9c7a399c6a08a38366ee02151ea04369b";
+    hash = "sha256-+z9IWgtc71GPPLDqNU4PXFDyD5Dczp3Bbrwzy/DaIts=";
+  };
+
+  preBuild = ''
+    cat > setup.py << EOF
+from setuptools import setup
+setup(scripts=['spotify-backup.py'])
+EOF'';
```

Comment:
it's unclear to me what this does, especially considering the whole derivation seems to just be copying a file to the output.

maybe add a comment for why this is needed?

Reply:
You're right, I did it that way because I started with the general python packaging template which assumes a setup.py existing.
I've switched to a simpler derivation after finding this approach: [https://stackoverflow.com/a/43837692](https://stackoverflow.com/a/43837692). I hope it's clearer this way.